lib/drivers/insuletDriver.js

```diff
@@ -677,7 +694,7 @@ module.exports = function (config) {
         wiz_idx = bolusrecs[b] + bolus.detail.calculation_record_offset;
       }
       var r = wizRecords[wiz_idx] || {};
-      if (bolus.detail.extended_duration_msec !== 0) {
+      if (bolus.detail.extended_duration_msec !== null) {
```

Comment:
This a debate we're going to have till the end of time! I like `_.isEmpty` for objects and arrays. I dislike it intensely for single values because I can never remember what its truth table actually is.

Also in this case we definitely *can't*, as the point is to distinguish between `null` and 0:
```
> _.isEmpty(0)
true
> _.isEmpty(null)
true
```